huaweicloud/services/vpc/resource_huaweicloud_vpc_traffic_mirror_filter.go

```diff
@@ -0,0 +1,196 @@
+package vpc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// API: VPC POST /v3/{project_id}/vpc/traffic-mirror-filters
+// API: VPC GET /v3/{project_id}/vpc/traffic-mirror-filters/{traffic_mirror_filter_id}
+// API: VPC PUT /v3/{project_id}/vpc/traffic-mirror-filters/{traffic_mirror_filter_id}
+// API: VPC DELETE /v3/{project_id}/vpc/traffic-mirror-filters/{traffic_mirror_filter_id}
+func ResourceTrafficMirrorFilter() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTrafficMirrorFilterCreate,
+		ReadContext:   resourceTrafficMirrorFilterRead,
+		UpdateContext: resourceTrafficMirrorFilterUpdate,
+		DeleteContext: resourceTrafficMirrorFilterDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func getTrafficMirrorFilterHttpUrl(d *schema.ResourceData, meta interface{}, isCreateResource bool) (string, *golangsdk.ServiceClient, error) {
```

Comment:
This function does so much that it returns not only the HTTP path, but also the client.
According to the function name, only the HTTP path should be returned.

Reply:
fixed